File: src/main/scala/joycai/utils/encrypt/AESHelper.scala
package joycai.utils.encrypt

import org.bouncycastle.util.encoders.Base64

trait AESHelper {

  def encrypt(content: Array[Byte]): Array[Byte]

  def encryptToString(content: Array[Byte]): String = Base64.toBase64String(encrypt(content))

  def decrypt(encryptedContent: Array[Byte]): Array[Byte]

  def decryptFromString(encryptedContent: String): Array[Byte] = decrypt(Base64.decode(encryptedContent))
}

File: src/main/java/joycai/utils/encrypt/AESHelper.java
package joycai.utils.encrypt;

import org.bouncycastle.util.encoders.Base64;

import javax.crypto.BadPaddingException;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.NoSuchPaddingException;
import java.security.InvalidAlgorithmParameterException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.security.NoSuchProviderException;

public interface AESHelper {
    byte[] encrypt(byte[] content) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException, BadPaddingException, IllegalBlockSizeException, NoSuchProviderException, InvalidAlgorithmParameterException;

   default String encryptToString(byte[] content) throws NoSuchPaddingException, InvalidAlgorithmParameterException, NoSuchAlgorithmException, IllegalBlockSizeException, BadPaddingException, NoSuchProviderException, InvalidKeyException {
        return Base64.toBase64String(encrypt(content));
    }
    byte[] decrypt(byte[] content) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException, BadPaddingException, IllegalBlockSizeException, NoSuchProviderException, InvalidAlgorithmParameterException;

    default byte[] decryptFromString(String encryptedContent) throws NoSuchPaddingException, InvalidAlgorithmParameterException, NoSuchAlgorithmException, IllegalBlockSizeException, BadPaddingException, NoSuchProviderException, InvalidKeyException {
        return decrypt(Base64.decode(encryptedContent));
    }
}
